xarray/test/test_dataset.py

```diff
@@ -323,6 +323,16 @@ def test_attr_access(self):
         self.assertEqual(ds.attrs['tmin'], -999)
         self.assertDataArrayIdentical(ds.tmin, ds['tmin'])
 
+    def test_dir(self):
+        ds = create_test_data()
+        # add a numbered key to ensure this doesn't break dir
+        ds[5] = list(ds.keys())[0]
+
+        some_expected_attrs = {'pipe', 'mean', 'isnull', 'var1',
+                               'dim1', 'numbers'}
```

Comment:
The plan (per mailing list discussino) is to drop Python 2.6 support for the next major release. That should probably be the next release, but we'll need to cut it out of the Travis-CI setup first.